### DISCUSSION TITLE
Multiple DS18B20 sensors with DS18x20_USE_ID_ALIAS in script

### DISCUSSION THREAD

**Discussion by marcohald** [arendst/Tasmota#17371] [Q&A] [answered]
### PROBLEM DESCRIPTION
It is not possisble to get the DS18B20 in a script with a Tasmota Build with DS18x20_USE_ID_ALIAS defined.
### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 12.1.1
  - [ ] Pre-compiled
  - [x] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Set the SetOption64 1 as described here https://github.com/arendst/Tasmota/issues/12935
Use this Script
```
>D
tempPumpe=10
tempKessel=10

>S
if upsecs%60==0
then
print tempPumpe %tempPumpe%
print tempKessel %tempKessel%
endif



>T
tempPumpe=DS18B20_DA07AA#Temperature
tempKessel=DS18B20-950A6B#Temperature
>W
@<p>Kessel: %tempKessel%<br></p>
@<p>Pumpe: %tempPumpe%<br></p>
```
with Tasmota build with this define
```
#define USE_SCRIPT_WEB_DISPLAY
#define USE_EEPROM
#define EEP_SCRIPT_SIZE 6200
#define USE_SCRIPT_JSON_EXPORT
#define DS18x20_USE_ID_AS_NAME
```
### EXPECTED BEHAVIOUR
The Variables tempPumpe and tempKessel should get the Sensor Readings and not stay at 10.
I tried the Variable Name with a - and with a _ as delimeter

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
The Webui shows the Sensors with this Names:
DS18B20-DA07AA Temperature	65.69 °C
DS18B20-950A6B Temperature	67.75 °C

### ADDITIONAL CONTEXT
I can provide the other Logs, but I think they are not specific to Scripting.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Transfering to discussions since it is a feature request

**Comment by sfromis** [comment]
Did you try `SetOption64`, which allows selection of sensor names with `_` instead of `-`? With scripting, it can only work if you do that.

**Comment by marcohald** [comment]
Yes I have it mentioned under how to reproduce:
Set the SetOption64 1 as described here https://github.com/arendst/Tasmota/issues/12935

**Comment by arendst** [comment, marked as answer]
Dump teleperiod message. I guess the names are different between GUI and JSON used by scripter

**Reply by marcohald** [reply]
```
19:52:06.908 MQT: tele/tasmota_5DE99E/SENSOR = {"Time":"2022-12-12T19:52:06","DS18B20-DA07AA":{"Id":"01143BDA07AA","Temperature":61.56},"DS18B20-950A6B":{"Id":"3C01F0950A6B","Temperature":69.31},"oktemp":1.00,"t1":0.00,"t2":0.00,"coff":0.00,"disconnects":0.00,"TempUnit":"C"}
```

After reviewing the SetOption64 it was turned off . I now turned it on again and did a reboot and it stays on.
The Script is now also working.
I have no Idea why the setting reverted back to 0.
Sorry for the trouble and thank you all for the help
